### PR TITLE
implement  generalised PaleyII symmetric conference matrices

### PR DESCRIPTION
**Pull request by dimpase**
Currently only symmetric conference matrices obtained from $GF(q)$ using Paley construction for $q\equiv 1\mod 4$
are used in building Hadamard matrices by PaleyII construction. But we can do better, use
any symmetric conference matrix known to Sage.

The 1st new to Sage example is of order 4*787

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by dimpase** [comment]
@MatteoCati - have you got around to look into this? If not, I can finish this myself.

**Comment by MatteoCati** [comment]
I've created a PR to add the code to this branch https://github.com/dimpase/sage/pull/7 since I could not edit the branch directly.
I (incorrectly) assumed you would have received a notification about it.

**Comment by dimpase** [comment]
Build documentation failure is a red herring, it seems.

@MatteoCati - ready for review

**Comment by dimpase** [comment]
@kwankyu - do you have an idea why CI docbuild fails here (at the place where html docs update is handled by `git merge`
(or maybe `git stash`) ? It looks like a bug in the CI.

**Comment by kwankyu** [comment]
Yes. The blocker fixes the bug in ci: https://github.com/sagemath/sage/pull/38468

You may merge the PR to build doc preview.

The blocker has been slumbering there for at least three weeks. That is why I am suggesting this: https://groups.google.com/g/sage-devel/c/ad9hQZw1PKQ

**Review by MatteoCati — APPROVED** [review]

File: src/sage/combinat/matrices/hadamard_matrix.py [hunk 1]
```diff
@@ -320,6 +320,87 @@ def hadamard_matrix_paleyII(n):
     return normalise_hadamard(H)
 
 
+def hadamard_matrix_from_symmetric_conference_matrix(n, existence=False, check=True):
+    r"""Construct a Hadamard matrix of order `n` from a symmetric conference matrix
+    of order `n/2`.
+
```
Comment (kwankyu) [inline src/sage/combinat/matrices/hadamard_matrix.py:327]:
```sage    
r"""
Construct a Hadamard matrix of order `n` from a symmetric conference matrix
of order `n/2`.
```
just to conform to the usual style.

File: src/sage/combinat/matrices/hadamard_matrix.py [hunk 2]
```diff
@@ -320,6 +320,87 @@ def hadamard_matrix_paleyII(n):
...
+        Traceback (most recent call last):
+        ...
+        ValueError: Cannot construct Hadamard matrix of order 64, a symmetric conference matrix of order 32 is not available in sage.
+        sage: hadamard_matrix_from_symmetric_conference_matrix(14)
+        Traceback (most recent call last):
+        ...
+        ValueError: No Hadamard matrix of order 14 exists.
+    """
+    if n < 0 or n % 4 != 0:
+        raise ValueError(f'No Hadamard matrix of order {n} exists.')
```
Comment (kwankyu) [inline src/sage/combinat/matrices/hadamard_matrix.py:384]:
```sage
raise ValueError(f'no Hadamard matrix of order {n} exists')
```
usual style of an error message.

File: src/sage/combinat/matrices/hadamard_matrix.py [hunk 3]
```diff
@@ -3279,6 +3371,8 @@ def symmetric_conference_matrix(n, check=True):
     - ``check`` -- boolean (default: ``True``); whether to check that output is
       correct before returning it. As this is expected to be useless, you may
       want to disable it whenever you want speed.
+    - ``existence`` -- boolean (default: ``False``); if true, only check that such
+      a matrix exists.
```
Comment (kwankyu) [inline src/sage/combinat/matrices/hadamard_matrix.py:3377]:
```sage
a matrix exists
```
no period at the end for one sentence description.

**Comment by kwankyu** [comment]
Otherwise lgtm.

File: src/sage/combinat/matrices/hadamard_matrix.py (hunk 3 above)
Comment (dimpase) [reply src/sage/combinat/matrices/hadamard_matrix.py:3377]: Sorry, note that there is a period in the previous item `check` as well - and this is how it's in most of Sage code I ever touched, IMHO.

Comment (dimpase) [reply src/sage/combinat/matrices/hadamard_matrix.py:3377]: the style of this file is that if there is a period at the end of an item, then it keeps so for the rest of the list.

File: src/sage/combinat/matrices/hadamard_matrix.py (hunk 2 above)
Comment (dimpase) [reply src/sage/combinat/matrices/hadamard_matrix.py:384]: It varies through this file. Note that changing this also needs doctest changes, I'd rather not do this here.

File: src/sage/combinat/matrices/hadamard_matrix.py (hunk 1 above)
Comment (dimpase) [reply src/sage/combinat/matrices/hadamard_matrix.py:327]: fixed by [4d56220](https://github.com/sagemath/sage/pull/38486/commits/4d562209ed2488da9ac60604a9ff48324218da18)

File: src/sage/combinat/matrices/hadamard_matrix.py (hunk 3 above)
Comment (kwankyu) [reply src/sage/combinat/matrices/hadamard_matrix.py:3377]: I don't agree. But okay as it is trivial.

File: src/sage/combinat/matrices/hadamard_matrix.py (hunk 2 above)
Comment (kwankyu) [reply src/sage/combinat/matrices/hadamard_matrix.py:384]: Though there are variance in the library, I insist on following the standard.

I would leave review of this PR to others.

File: src/sage/combinat/matrices/hadamard_matrix.py (hunk 1 above)
Comment (kwankyu) [reply src/sage/combinat/matrices/hadamard_matrix.py:327]: Thanks.

**Comment by kwankyu** [comment]
I think you can set this to "positive review" since @MatteoCati implicitly acknowledged it.

**Comment by dimpase** [comment]
Thanks!

**Pull request merged**